flaml/automl/automl.py

```diff
@@ -54,17 +55,36 @@
 from flaml.automl.training_log import training_log_reader, training_log_writer
 from flaml.default import suggest_learner
 from flaml.version import __version__ as flaml_version
+from flaml.spark.utils import check_spark
 
 logger = logging.getLogger(__name__)
 logger_formatter = logging.Formatter(
     "[%(name)s: %(asctime)s] {%(lineno)d} %(levelname)s - %(message)s", "%m-%d %H:%M:%S"
 )
+logger.propagate = False
 
 try:
     import mlflow
 except ImportError:
     mlflow = None
 
+try:
+    import pyspark
+    from pyspark.sql import SparkSession
+
+    spark_import = True
```

Comment:
A better name for these might be `spark_available` and `ray_available`

Reply:
I think `spark_import` and `ray_import` is more accurate. As this import code will pass if pyspark or ray is installed. However, a real spark cluster or ray cluster is not yet set up. That's why we have `check_spark` in spark.utils.